packages/web-component/src/lib/helpers/templates.ts

```diff
@@ -89,16 +90,15 @@ const replaceProvisionURL = (
 };
 
 /**
- * Perform action in base element based on screen state
+ * Update a screen template based on the screen state
  *  - Show/hide error messages
- *  - Replace values of element inputs with screen state's inputs
  *  - Replace element templates ({{...}} syntax) with screen state object
  */
-export const replaceWithScreenState = (
+export const updateTemplateFromScreenState = (
```

Comment:
what is the case of calling `updateTemplateFromScreenState` without `updateScreenFromScreenState`?

also noting that names are a bit confusing to me, the `template` vs `screen` terminology is not that obvious for me, maybe we can call this one `updateScreenFromScreenState` and the bellow `updateInputsFromScreenState`

Reply:
> what is the case of calling `updateTemplateFromScreenState` without `updateScreenFromScreenState`?

Both fn should be called, just in a different timing

> also noting that names are a bit confusing to me, the `template` vs `screen` terminology is not that obvious for me, maybe we can call this one `updateScreenFromScreenState` and the bellow `updateInputsFromScreenState`

IMO `updateInputsFromScreenState` is a bit misleading, because we are not updating only inputs
One updates the template of the screen, before it's mounted, the second, update the screen, after the template is already mounted

Reply:
> Both fn should be called, just in a different timing


gotcha, maybe one should be called from the other and not to expose both to the caller, not critical